fe/fe-core/src/main/java/org/apache/doris/rewrite/ExtractCommonFactorsRule.java

```diff
@@ -462,6 +464,13 @@ private Expr rewriteOrToIn(List<Expr> exprs) {
         boolean isOrToInAllowed = true;
         Set<String> slotSet = new LinkedHashSet<>();
 
+        int rewriteThreshold;
+        if (ConnectContext.get() == null) {
+            rewriteThreshold = 2;
```

Comment:
is default value 2 too small?

Reply:
Default value is set to 2, it means rewrite will always be allowed.